File: Android/app/src/main/java/com/etiennebaudoux/clipboardzanager/models/Thumbnail.java
package com.etiennebaudoux.clipboardzanager.models;

import com.etiennebaudoux.clipboardzanager.enums.ThumbnailDataType;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

/**
 * Represents a thumbnail of a data of the clipboard.
 */
public class Thumbnail implements Serializable {
    //region Properties

    //region Value

    @SerializedName("Value")
    private String _value;

    /**
     * Gets the value of the thumbnail.
     *
     * @return
     */
    public String getValue() {
        return _value;
    }

    /**
     * Sets the value of the thumbnail.
     *
     * @param value
     */
    public void setValue(String value) {
        _value = value;
    }

    //endregion

    //region Type

    @SerializedName("Type")
    private ThumbnailDataType _type;

    /**
     * Gets the type of the value.
     *
     * @return
     */
    public ThumbnailDataType getType() {
        return _type;
    }

    /**
     * Sets the type of the value.
     *
     * @param value
     */
    public void setType(ThumbnailDataType value) {
        _type = value;
    }

    //endregion

    //endregion
}

File: ClipboardZanager/Sources/ClipboardZanager.Core.Desktop/Models/Thumbnail.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using ClipboardZanager.Core.Desktop.ComponentModel;
using ClipboardZanager.Core.Desktop.Enums;

namespace ClipboardZanager.Core.Desktop.Models
{
    /// <summary>
    /// Represents a thumbnail of a data of the clipboard.
    /// </summary>
    [Serializable]
    internal sealed class Thumbnail
    {
        [NonSerialized]
        private AsyncObservableCollection<string> _fileList;

        /// <summary>
        /// Gets or sets the value of the thumbnail.
        /// </summary>
        public string Value { get; set; }

        /// <summary>
        /// Gets or sets the type of the value.
        /// </summary>
        public ThumbnailDataType Type { get; set; }

        /// <summary>
        /// If the value is a file list, gets one of the 3 files.
        /// </summary>
        /// <param name="index">The index, between 0 and 2</param>
        /// <returns>The file path</returns>
        internal string GetFilePath(int index)
        {
            if (index > 2)
            {
                throw new ArgumentOutOfRangeException(nameof(index));
            }

            var files = GetFilesPath();

            if (files?.Count > index)
            {
                return files[index];
            }

            return string.Empty;
        }

        /// <summary>
        /// If the value is a file list, gets the list of file.
        /// </summary>
        /// <returns>The list of file path</returns>
        internal AsyncObservableCollection<string> GetFilesPath()
        {
            if (Type != ThumbnailDataType.Files)
            {
                return null;
            }

            if (_fileList == null)
            {
                _fileList = new AsyncObservableCollection<string>(DataHelper.FromBase64<List<string>>(Value));
            }

            return _fileList;
        }
    }
}
